components/notebook-controller/pkg/util/util.go

```diff
@@ -79,3 +80,27 @@ func CopyServiceFields(from, to *corev1.Service) bool {
 
 	return requireUpdate
 }
+
+func CopyVirtualService(from, to *unstructured.Unstructured) bool {
+	fromSpec, found, err := unstructured.NestedMap(from.Object, "spec")
+	if !found {
+		return true
+	}
+	if err != nil {
+		return true
+	}
+
+	toSpec, found, err := unstructured.NestedMap(to.Object, "spec")
+	if !found {
+		return true
```

Comment:
we should not return here.
We should still SetNestedMap to copy spec from "from" to "to" right?

Reply:
done.